### ISSUE TITLE
Launch of ScriptHook crashes the game

### ISSUE THREAD

**Issue by Zorakie** [MartinJK/Mafia-Definitive-Edition-ScriptHook#1]
Hi,
whenever I try to launch the ScriptHook (the game is running), the game just crashes. Unfortunately, even logs don't tell anything understandable.

Using the EPIC games version, if that matters.

Administrator rights for game and/or Scripthook seems to not have any effect.

Steps:
Launch Mafia through exe (I have disabled the original launcher), load into the game (in my case Free ride, launch ScriptHook exe, the game crashes.

Game exe: version 1.0.0.1
Installed scripts: None - initially tried NoClip, but after crashes, I have just removed any script to check just scripthook load

Scripthook log:

> M1DEScriptHook::M1DEScriptHook
> PluginSystem::LoadPlugins
> M1DEScriptHook::StartThreads
> LuaStateManager::StartThread
> M1DEScriptHook::mainThread
> LuaFunctions::LuaFunctions
> LuaFunctions::LuaFunctions Game is not ready, script engine not initialized, retry
> LuaFunctions::LoadPointers
> m_pMainGameScriptMachine (146312da0)
> lua_pcall (14495c6e9)
> lua_tostring (140e4dd50)
> lua_isstring (142fa4545)
> lua_loadBuffer (144919a90)
> lua_newthread (14490f570)
> lua_pushcclosure (144911a10)
> lua_setfield (140e4db10)
> lua_setglobal (143d76530)
> LuaFunctions::LoadPointers Finished
> LuaStateManager::StateChanged
> LuaFunctions::Setup
> LuaFunctions::Setup::<lambda_3f731b9485bcd7f28ee7d759df59c8b8>::operator ()

Any idea, what is going on?

Thank you and have a nice day!

**Comment by Fastmancz** [comment]
I have the same problem at Epic Games store. :(

**Comment by WolverStones** [comment]
The script hook only works on the steam version of the game

**Comment by MartinJK** [comment]
Hey guys. Thanks for getting in touch. In fact, the game only supports the Steam version at the moment. Support for EPIC coming within the next days

**Comment by vecuccio** [comment]
It seems pretty flaky with the Steam version too.  Using the 'Better NoClip' often causes the game to CTD. I keep checking back every day or so to see if there's an update.

**Comment by MartinJK** [comment]
> It seems pretty flaky with the Steam version too. Using the 'Better NoClip' often causes the game to CTD. I keep checking back every day or so to see if there's an update.

Yes, can confirm that. There's a complete overhaul of the mod in the works, which will be stable (without crashes), feature steam/epic support and many more features too :) 

We're working hard to release it this week. I'll keep you posted.

**Comment by MartinJK** [comment]
Hey guys, 

gome some nice update for you. On Tuesday, October 6th NOMAD Group is going  to release the overhaul for the Mafia ScriptHook with a ton of new features. This GitHub project is deprecated from now on, for more information visit

https://community.nomad-group.net/index.php?/blogs/entry/10-development-blog-10-mplus-mafia-definitive-edition-scripthook/

https://www.youtube.com/watch?v=pQ9HM20a83g
